File: Java/lab12_Flyweight/task_2_2/src/Wheel.java
public class Wheel {
    public enum Material {
        Steel, //  стальний диск
        Alloy  //  легкосплавний
    }

    final private Material material;
    final int diameter;

    public Wheel(int diameter) {
        this.material = Material.Steel;
        this.diameter = diameter;
    }

    @Override
    public String toString() {
        return "Wheel{" +
                "material=" + material +
                ", diameter=" + diameter +
                '}';
    }
}

File: Cpp/lab12_Flyweight/task_2_2/src/wheel.cpp
#include "wheel.hpp"


Wheel::Wheel(int diameter)
: diameter(diameter), material(Wheel::Material::Steel) {}

std::ostream &operator<<(std::ostream& os, Wheel const& wheel)
{
  os << "Wheel{material=";

  switch (wheel.material)
  {
    case Wheel::Material::Steel:
      os << "Steel";
      break;
    case Wheel::Material::Alloy:
      os << "Alloy";
      break;
  }

  os << ", diameter=" << wheel.diameter << "}";

  return os;
}
